New Term Entries: Python Dictionary Methods (#728)
Codecademy/docs: content/python/concepts/dictionaries/terms/setdefault/setdefault.md

---
Title: '.setdefault()'

> Description: "Returns the value of a specified key. If the key doesn't exist, it is inserted with the specified value."

Reviewer: ```suggestion
Description: 'Returns the value of a specified key. If the key doesn't exist, it is inserted with the specified value.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Dictionaries'
  - 'Methods'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The `.setdefault()` method will return the value of a dictionary entry for a specified key. If the specified key does not exist in the dictionary, it will create an entry for it with the specified value.

## Syntax

```pseudo

> dictionary.setdefault(key,value)

Reviewer: ```suggestion
dictionary.setdefault(key, value)
```

```

Where `key` is the key of the value to return. If `key` does not exist in `dictionary`, it will be created. `value` is an optional parameter specifying the value for `key` if a new entry is created, it has no effect otherwise. If `value` is not specified, the value `None` will be used.

## Codebyte Example

> The following example creates a dictionary, then uses `.setdefault()` to return values from it.

Reviewer: ```suggestion
The following example creates a dictionary, then uses `.setdefault()` to return values from it:
```
